source/views/building-hours/stateful-list.js

```diff
@@ -92,24 +117,51 @@ export class BuildingHoursView extends React.Component<Props, State> {
       buildings = defaultData.data
     }
     this.setState(() => ({
-      buildings: groupBuildings(buildings),
+      buildings: loadAllBuildings(buildings, this.props.favoriteBuildings),
       now: moment.tz(CENTRAL_TZ),
     }))
   }
 
+  onUpdate = () => {
+    this.fetchData()
+    this.props.onSaveFavorites(this.props.favoriteBuildings)
```

Comment:
This is the other half of my comment about moving the "make a favorite" logic into the redux file; when you do that, you won't need to save in onUpdate at all, because the update will have happened as soon as the heart button was pressed.